diff -urN nss-mysql-0.35.org/acinclude.m4 nss-mysql-0.35/acinclude.m4
--- nss-mysql-0.35.org/acinclude.m4	Tue Mar 26 23:18:57 2002
+++ nss-mysql-0.35/acinclude.m4	Tue Mar 26 23:20:06 2002
@@ -91,9 +91,9 @@
 fi
 ])
 
-if test ! -f "${MYSQL_LIB}/libmysqlclient.a"
+if test ! -f "${MYSQL_LIB}/libmysqlclient.so"
 then
-	if test -f "${MYSQL_HOME}/lib/libmysqlclient.a"
+	if test -f "${MYSQL_HOME}/lib/libmysqlclient.so"
 	then
 		MYSQL_LIB=${MYSQL_HOME}/lib
 	else
